dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreUtils.cs

```diff
@@ -56,7 +56,7 @@ public static async IAsyncEnumerable<T> WrapAsyncEnumerableAsync<T>(
             var more = await enumerator.MoveNextAsync();
             return (enumerator.Current, more);
         }
-        catch (Exception ex)
+        catch (Exception ex) when (ex is not (NotSupportedException or ArgumentException))
```

Comment:
I think we should be going the other way, and specifically identifying the exceptions we want to wrap. For example, this will still wrap OperationCanceledException.

I'd recommend not dealing with this in this PR, but doing a proper separate pass across all providers to check on the exception behavior, as discussed offline (#11512).